### ISSUE TITLE
[clang-tidy] modernize-use-override, add an option to allow both virtual and override keywords

### ISSUE THREAD

**Issue by berkaysahiin** [llvm/llvm-project#202030]
Having an `override` keyword when overriding a virtual function is really valuable. No doubt on that. 

But one may prefer repeating the `virtual` keyword as well as using the `override` keyword (stylistic preferences). Arguably, it is easier spot `virtual` keyword compare to `override` keyword as `virtual` is usually the first keyword used on a method signature.

Currently, there is no way to configure [modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html) to have such behavior:
- If I am overriding a virtual function, but there is no `override`, warn on missing `override`
- Do not warn on `virtual` if I have used `override` alongside with it.

```cpp

virtual void Foo() override; // Should be okay.

```

**Comment by zeyi2** [comment]
Hi @berkaysahiin, are you already working on this issue? If not, I’d be happy to take it on :)

**Comment by berkaysahiin** [comment]
All yours @zeyi2

**Comment by vbvictor** [comment]
> Currently, there is no way to configure [modernize-use-override](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html) to have such behavior:
>
> If I am overriding a virtual function, but there is no override, warn on missing override
Do not warn on virtual if I have used override alongside with it.
virtual void Foo() override; // Should be okay.

Isn't this option https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-override.html#cmdoption-arg-AllowOverrideAndFinal what you need?

**Comment by berkaysahiin** [comment]
From what I understand from `AllowOverrideAndFinal` option, it seems not. There is no `final` usage in the case I am trying to explain: https://godbolt.org/z/43T6c67Ea

**Comment by vbvictor** [comment]
> From what I understand from `AllowOverrideAndFinal` option, it seems not. There is no `final` usage in the case I am trying to explain: https://godbolt.org/z/43T6c67Ea

Oh right, nevermind me, must have been out of my mind.

**Comment by vbvictor** [comment]
Maybe someone tried to implement this in the past but PR never landed

**Comment by berkaysahiin** [comment]
Ah, I didn't check that. That's on me.

#144916 this ?

**Comment by vbvictor** [comment]
Yes, must be

**Comment by carlosgalvezp** [comment]
FWIW it's recommended to only have one specifier (but one can of course choose to not follow these guidelines):

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#rh-override
